OWML.ModHelper.Menus/ModInputCombinationElement.cs

```diff
@@ -0,0 +1,171 @@
+            YesButton.Title = "Edit";
+            YesButton.OnClick += OnEditClick;
+
+            var deleteCommand = new SingleAxisCommand();
+            var deleteBindingGmpd = new InputBinding(JoystickButton.FaceUp);
+            var deleteBindingKbrd = new InputBinding(KeyCode.Delete);
+            deleteCommand.SetInputs(deleteBindingGmpd, deleteBindingKbrd);
+            commandObject = new GameObject();
+            var updater = commandObject.AddComponent<ModCommandUpdater>();
+            updater.Initialize(deleteCommand);
+            commandComponent = commandObject.AddComponent<ModCommandListener>();
+            commandComponent.Initialize(deleteCommand);
+            commandComponent.OnNewlyReleased += OnDeleteButton;
+            NoButton.Title = "Delete";
+            NoButton.OnClick += OnDeleteClick;
+
+            Initialize(menu);
+            layoutGroup.childControlWidth = false;
+            layoutGroup.childControlHeight = false;
+            layoutGroup.childForceExpandHeight = false;
+            layoutGroup.childForceExpandWidth = false;
+            layoutGroup.spacing = 0f;
+            var constantGraphics = _layoutObject.GetComponentsInChildren<Graphic>(true);
+            _layoutObject.transform.GetChild(1).gameObject.SetActive(false);
+            Layout = new LayoutManager(layoutGroup, MonoBehaviour.FindObjectOfType<UIStyleManager>(),
+                ModUIStyleApplier.ReplaceStyleApplier(toggle.gameObject), scale, constantGraphics);
+            UpdateContents();
+            _popupMenu = popupMenu;
+        }
+
+        private void UpdateContents()
+        {
+            Layout.Clear();
+            if (_combination != "")
+            {
+                var keyStrings = _combination.Split('+');
+                for (var j = 0; j < keyStrings.Length; j++)
+                {
+                    AddKeySign(keyStrings[j]);
+                    if (j < keyStrings.Length - 1)
+                    {
+                        Layout.AddTextAt("+", Layout.ChildCount - 1);
+                    }
+                }
+            }
+            Layout.UpdateState();
+        }
+
+        private void AddKeySign(string key)
+        {
+            Layout.AddPictureAt(
+               _inputHandler.Textures.KeyTexture(key)
+                , Layout.ChildCount - 1, ScaleDown);
+        }
+
+        private void OnEditButton()
+        {
+            if (Toggle.GetValue<bool>("_amISelected"))
+            {
+                OnEditClick();
+            }    
+        }
+
+        private void OnEditClick()
+        {
+            EventSystem.current.SetSelectedGameObject(this.Toggle.gameObject);//make sure it gets selected after popup closes
+
+            _popupMenu.OnConfirm += OnPopupMenuConfirm;
+            _popupMenu.OnCancel += OnPopupMenuCancel;
+            _popupMenu.Open(_combination, (Menu is IModInputCombinationMenu) ? (Menu as IModInputCombinationMenu).Title : "",
```

Comment:
the title can be a variable to reduce the line length

Reply:
K